graylog2-web-interface/src/views/components/aggregationwizard/elementConfiguration/GroupByConfiguration.tsx

```diff
@@ -73,21 +73,22 @@ const GroupByConfiguration = () => {
       )}
       <FieldArray name="groupBy.groupings"
                   render={() => (
-                    <>
-                      <div>
-                        {groupBy?.groupings?.map((grouping, index) => {
-                          return (
-                            // eslint-disable-next-line react/no-array-index-key
-                            <ElementConfigurationContainer key={`grouping-${index}`}
-                                                           onRemove={() => removeGrouping(index)}
-                                                           elementTitle={GroupByElement.title}>
-                              <GroupBy index={index} />
-                            </ElementConfigurationContainer>
-                          );
-                        })}
-                      </div>
-                    </>
+                    <SortableList items={groupBy?.groupings}
+                                  onMoveItem={(newGroupings) => setFieldValue('groupBy.groupings', newGroupings)}
+                                  customListItemRender={({ index, dragHandleProps, draggableProps, className, ref }) => (
+                                    <ElementConfigurationContainer key={`grouping-${index}`}
```

Comment:
Wouldn't it be better to include the random id in the key, instead of the array index? The array index is prone to changes when reordering the list, but the random id is stable?

Reply:
You are right, we should include the id now where it is possible.